src/ServiceControl.Infrastructure/IntegratedSetup.cs

```diff
@@ -52,7 +52,12 @@ public static Task Run()
             process.BeginOutputReadLine();
             process.BeginErrorReadLine();
 
-            return process.WaitForExitAsync();
+            await process.WaitForExitAsync().ConfigureAwait(false);
+
+            if (process.ExitCode != 0)
+            {
+                Environment.Exit(process.ExitCode);
```

Comment:
I'm wondering if it would be better to just return the exit code from this method instead, and let then let the caller `return exitCode` when non-zero.